pkg/operation/botanist/component/vpnshoot/vpnshoot.go

```diff
@@ -0,0 +1,515 @@
+}
+
+func (v *vpnShoot) computeResourcesData() (map[string][]byte, error) {
+
+	var (
+		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+		secretTLSAuth = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vpn-shoot-tlsauth",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: v.secrets.TLSAuth.Data,
+		}
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				//TODO(shafeeqes): what about for ReversedVPN not enabled case?
+				Name:      "vpn-shoot",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: v.secrets.Server.Data,
+		}
+		secretDH *corev1.Secret
+	)
+
+	utilruntime.Must(kutil.MakeUnique(secretTLSAuth))
+	utilruntime.Must(kutil.MakeUnique(secret))
+
+	if !v.values.ReversedVPNEnabled {
+		secretDH = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vpn-shoot-dh",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: v.secrets.DH.Data,
+		}
+		utilruntime.Must(kutil.MakeUnique(secretDH))
+	}
+
+	var (
+		intStrMax  = intstr.FromString("100%")
+		intStrZero = intstr.FromString("0%")
+
+		vpa                *autoscalingv1beta2.VerticalPodAutoscaler
+		service            *corev1.Service
+		clusterRole        *rbacv1.ClusterRole
+		clusterRoleBinding *rbacv1.ClusterRoleBinding
+
+		serviceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vpn-shoot",
+				Namespace: metav1.NamespaceSystem,
+				Labels:    map[string]string{v1beta1constants.LabelApp: LabelValue},
+			},
+		}
+		networkPolicy = &networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "gardener.cloud--allow-vpn",
+				Namespace: metav1.NamespaceSystem,
+				Annotations: map[string]string{
+					v1beta1constants.GardenerDescription: "Allows the VPN to communicate with shoot components and makes the VPN reachable from the seed.",
+				},
+			},
+			Spec: networkingv1.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{v1beta1constants.LabelApp: LabelValue},
+				},
+				Egress:      []networkingv1.NetworkPolicyEgressRule{{}},
+				Ingress:     []networkingv1.NetworkPolicyIngressRule{{}},
+				PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress, networkingv1.PolicyTypeIngress},
+			},
+		}
+
+		deployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deploymentName,
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					v1beta1constants.GardenRole:     v1beta1constants.GardenRoleSystemComponent,
+					v1beta1constants.LabelApp:       LabelValue,
+					managedresources.LabelKeyOrigin: managedresources.LabelValueGardener,
+				},
+			},
+			Spec: appsv1.DeploymentSpec{
+				RevisionHistoryLimit: pointer.Int32(1),
```

Comment:
```suggestion
				RevisionHistoryLimit: pointer.Int32(2),
```

Nit: Set this to two. I know, it was one in the Helm chart, but we actually want to use `2` for most of our components.